src/Middleware/ThrottleMiddleware.php

```diff
@@ -0,0 +1,218 @@
+<?php
+namespace Muffin\Throttle\Middleware;
+
+use Cake\Cache\Cache;
+use Cake\Core\InstanceConfigTrait;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Zend\Diactoros\Stream;
+
+class ThrottleMiddleware
+{
+
+    use InstanceConfigTrait;
+
+    /*
+     * Default config for Throttle Middleware
+     *
+     * @var array
+     */
+    protected $_defaultConfig = [
+        'message' => 'Rate limit exceeded',
+        'interval' => '+1 minute',
+        'limit' => 10,
+        'headers' => [
+            'limit' => 'X-RateLimit-Limit',
+            'remaining' => 'X-RateLimit-Remaining',
+            'reset' => 'X-RateLimit-Reset'
+        ]
+    ];
+
+    /**
+     * Cache configuration name
+     *
+     * @var string
+     */
+    public static $cacheConfig = 'throttle';
+
+    /**
+     * Cache expiration key suffix
+     *
+     * @var string
+     */
+    public static $cacheExpirationSuffix = 'expires';
+
+    /**
+     * Unique client identifier
+     *
+     * @var string
+     */
+    protected $_identifier;
+
+    /**
+     * Number of connections after increment
+     *
+     * @var string
+     */
+    protected $_count;
+
+    /**
+     * ThrottleMiddleware constructor.
+     *
+     * @param array $config Configuration options
+     */
+    public function __construct($config = [])
+    {
+        $config += [
+            'identifier' => function (ServerRequestInterface $request) {
+                return $request->clientIp();
+            }
+        ];
+        $this->setConfig($config);
+    }
+
+    /**
+     * Called when the class is used as a function
+     *
+     * @param \Psr\Http\Message\ServerRequestInterface $request Request object
+     * @param \Psr\Http\Message\ResponseInterface $response Response object
+     * @param callable $next Next class in middelware
+     * @return \Psr\Http\Message\ResponseInterface
+     */
+    public function __invoke(ServerRequestInterface $request, ResponseInterface $response, callable $next)
+    {
+        $this->_setIdentifier($request);
+        $this->_initCache();
+        $this->_count = $this->_touch();
+
+        if ($this->_count > $this->getConfig('limit')) {
+            $stream = new Stream('php://memory', 'wb+');
+            $stream->write((string)$this->getConfig('message'));
+
+            return $response->withStatus(429)
+                ->withBody($stream);
+        }
+
+        $response = $next($request, $response);
```

Comment:
These last 2 statements should be swapped. Calling the next middleware in queue should be the last statement.